src/index.ts

```diff
@@ -18,24 +18,30 @@ export default {
 		env: Env,
 		ctx: ExecutionContext
 	): Promise<Response> {
-		let formJson = { "Title": "", "Description": "", "Date due": "" };
+		let formJson = { "Title": "", "Description": "", "Date due": "", "projectId": "", "listId": "" };
 		formJson = await request.json();
+		if (formJson["Date due"] === "") {
```

Comment:
good, mostly. this can be shortened a lot though... try this on line 31 instead. you won't need 23-27 if this works.

`due_on: formJson["Date Due"] ?? new Date().getDate() + 7`

this means  `take formJson["Date Due"] unless undefined or null, in which case take `new Date().getDate() + 7`